apb_devel/index.adoc

```diff
@@ -0,0 +1,201 @@
+and provides a tutorial and reference material on writing your own APBs.
+
+[[apb-devel-intro-design]]
+== Design Overview
+
+An APB is a lightweight application definition that borrows several concepts
+from the link:https://github.com/projectatomic/nulecule[Nulecule] and
+link:http://www.projectatomic.io/docs/atomicapp/[Atomicapp] projects, namely the
+concept of a short-lived container with the sole purpose of orchestrating the
+deployment of the intended application. For the case of APBs, this short-lived
+container is the APB itself: a container with an
+link:https://www.ansible.com/[Ansible] runtime environment plus any files
+required to assist in orchestration, such as playbooks, roles, and extra
+dependencies.
+
+The
+xref:../architecture/service_catalog/ansible_service_broker.adoc#arch-ansible-service-broker[OpenShift Ansible broker] (OAB) is an implementation of the Open Service Broker (OSB) API
+that manages applications defined by APBs. The OAB is supported and deployed by
+default starting in {product-title} 3.7.
+
+Specification of an APB is intended to be lightweight, consisting of several
+named playbooks and a metadata file to capture information such as parameters to
+pass into the application.
+
+[[apb-devel-intro-workflow]]
+== Workflow
+
+The APB workflow is broken up into the following steps:
+
+. Preparation
+.. APB initialization
+.. APB spec file
+.. Actions (provision, deprovision, bind, unbind)
+. Build
+. Deploy
+
+[[apb-devel-intro-prep]]
+=== Preparation
+
+[[apb-devel-intro-apb-init]]
+==== APB Initialization
+
+////
+image::apb-prepare.png[]
+////
+
+The first step to creating an APB is to run the `apb init` command, which
+creates the required skeleton directory structure and a few required files (for
+example, the *_apb.yml_* spec file) for the APB.
+
+The following shows an example directory structure of an APB:
+
+[[apb-directory-structure]]
+.Directory Structure
+----
+example-apb/
+├── Dockerfile
+├── apb.yml
+└── roles/
+│   └── example-apb-openshift
+│       ├── defaults
+│       │   └── main.yml
+│       └── tasks
+│           └── main.yml
+└── playbooks/
+    └── provision.yml
+    └── deprovision.yml
+    └── bind.yml
+    └── unbind.yml
+----
+
+[[apb-devel-intro-spec-file]]
+==== APB Spec File
+
+An APB spec file (*_apb.yml_*) must be edited for your specific application. For
+example, the `etherpad-apb` spec file looks as follows:
+
+----
+name: fusor/etherpad-apb
+description: Note taking web application
+bindable: true
+async: optional
+metadata:
+  displayName: Etherpad APB
+parameters:
+  - name: hostport
+    description: The host TCP port as the external endpoint
+    type: int
+    default: 9001
+  - name: db_user
+    description: Database User
+    type: string
+----
+
+For an APB that does not have any parameters, the `parameters` field should be
+blank:
+
+----
+parameters: []
+----
+
+The `metadata` field is optional and used when integrating with the
+{product-title} service catalog.
```

Comment:
I think we may have Service Catalog capitalized as a proper noun elsewhere, but I could be wrong.

Reply:
Just checked and I have it capitalized in my UX update PR. We can discuss whether it should or should not be and then add it to the glossary.